### PR TITLE
Poc improved

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# POC IMPROVED

Similar to the POC posted in microdata-testing, but improved with:
* testing with good coverage
* pydantic request models
* the gunicorn logger from microdata-pseudonym-service
* atomic operation for writing of new add_data jobs that will stop any datasets from being imported twice in a row

### CODE REVIEW

**Review by vaskalan — APPROVED** [review]
Looks good!

File: job_service/repository/job_db.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                {
+                    "$push": {
+                        "logs": {"at": now, "message": log}
+                    }
+                }
+            )
+
+    def __stringify_datetime_logs(self, logs):
+        for log in logs:
+            log["at"] = log["at"].strftime("%Y/%m/%d, %H:%M:%S")
```
Comment (pawbu) [inline job_service/repository/job_db.py:129]: Use a standard ISO 8601 format? Then it's up to the client to parse it/display it as it finds suitable.

https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat

>>> now = datetime.now()
>>> now.isoformat()
'2022-04-06T14:06:08.959432'

File: test/repository/test_job_db.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+PROCESSED_JOB = {
+    'jobId': '123-123-123-123',
+    'logs': [
+        {'at': '2020/01/01, 00:00:00', 'message': 'mock'}
+    ],
+    'status': 'queued'
+}
+
+
+class MockedJobsCollection:
```
Comment (pawbu) [inline test/repository/test_job_db.py:37]: Nice

**Review by pawbu — COMMENTED** [review]
Well done!
I have some questions & comments

File: job_service/repository/job_db.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    def get_job(self, job_id: str) -> dict:
+        """
+        Returns job with matching job_id from database.
+        Raises NotFoundException if no such job is found.
+        """
+        job = self.db.jobs.find_one({"jobId": job_id})
+        if job is None:
+            raise NotFoundException(f"No job found for jobId: {job_id}")
+
+        del job["_id"]
```
[the remark this answers is not in the document]
Comment (DanielElisenberg) [reply job_service/repository/job_db.py:38]: _id is a field mongodb sets for indexing porpuses and serves as the documents primary key (It is generated so its naturally indexed by time created as well which is pretty clever)
This _id is auto generated by mongodb so we are not really interested in exposing it beyond that.

File: job_service/repository/job_db.py (hunk 1 above)
Comment (DanielElisenberg) [reply job_service/repository/job_db.py:129]: this is smart:100: Let's add this!

**Pull request merged**